Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporterUtils.h"
+#include "TypeConversions.h"
+#include <AzCore/Asset/AssetManager.h>
+#include <AzCore/Asset/AssetManagerBus.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzCore/std/string/regex.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+
+namespace ROS2
+{
+
+    bool Utils::IsWheelURDFHeuristics(const urdf::LinkConstSharedPtr& link)
+    {
+        const AZStd::regex wheel_regex("wheel[_]||[_]wheel");
+        const AZStd::regex joint_regex("(?i)joint");
+        const AZStd::string link_name(link->name.c_str(), link->name.size());
+        AZStd::smatch match;
+        // check is name is catchy for wheel
+        if (!AZStd::regex_search(link_name, match, wheel_regex))
+        {
+            return false;
+        }
+        // but it should cointain joint word
+        if (AZStd::regex_search(link_name, match, joint_regex))
+        {
+            return false;
+        }
+        // wheel need to have collision and visuals
+        if (!(link->collision && link->visual))
+        {
+            return false;
+        }
+        // and finally parent joint needs to be CONTINOUS
+        if (link->parent_joint && link->parent_joint->type == urdf::Joint::CONTINUOUS)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    AZ::Transform Utils::GetWorldTransformURDF(const urdf::LinkSharedPtr& link, AZ::Transform t)
+    {
+        if (link->getParent() != nullptr)
+        {
+            t = URDF::TypeConversions::ConvertPose(link->parent_joint->parent_to_joint_origin_transform) * t;
+            return GetWorldTransformURDF(link->getParent(), t);
+        }
+        return t;
+    }
+
+    AZStd::unordered_map<AZStd::string, urdf::LinkSharedPtr> Utils::GetAllLinks(const std::vector<urdf::LinkSharedPtr>& childLinks)
+    {
+        AZStd::unordered_map<AZStd::string, urdf::LinkSharedPtr> pointers;
+        std::function<void(const std::vector<urdf::LinkSharedPtr>&)> link_visitor =
+            [&](const std::vector<urdf::LinkSharedPtr>& child_links) -> void
+        {
+            for (const urdf::LinkSharedPtr& child_link : child_links)
+            {
+                AZStd::string link_name(child_link->name.c_str(), child_link->name.size());
+                pointers[link_name] = child_link;
+                link_visitor(child_link->child_links);
+            }
+        };
+        link_visitor(childLinks);
+        return pointers;
+    }
+
+    AZStd::unordered_map<AZStd::string, urdf::JointSharedPtr> Utils::GetAllJoints(const std::vector<urdf::LinkSharedPtr>& childLinks)
+    {
+        AZStd::unordered_map<AZStd::string, urdf::JointSharedPtr> joints;
+        std::function<void(const std::vector<urdf::LinkSharedPtr>&)> link_visitor =
```

Comment:
repeat: AZStd::function can be used here.
```suggestion
        AZStd::function<void(const std::vector<urdf::LinkSharedPtr>&)> link_visitor =
```